Cython/Compiler/FusedNode.py

```diff
@@ -140,6 +140,18 @@ def copy_cdef(self, env):
         for cname, fused_to_specific in permutations:
             copied_node = copy.deepcopy(self.node)
 
+            # unlike for the argument types, specializing the return type can fail, so tentatively
+            # try it here and produce an error message if it fails.
+            # specialize can also go through layers of pointers/const which makes anything other
+            # than "try...except" difficult
```

Comment:
That suggests to me that this might not be the right place to do this. Can't we detect this earlier? E.g. when analysing the function signature?

Reply:
It probably could be done earlier but I think the problem would be the same - to work out if it specializes then you have to duplicate most of the specialization logic in PyrexTypes.

I wonder if it'd be better if `specialize` returned `error_type` rather than raising an exception to allow for "safe" checks that something specializes?

Reply:
Isn't it an issue whenever a fused type occurs in the return type that isn't used in any of the arguments? Or are there legitimate use cases for that as well?